src/test/javascript/esutil_test.ts

```diff
@@ -56,99 +55,35 @@ suite('getEventComments', () => {
       {
         description: 'This is an event',
         name: 'event-name',
-        params: [
-          {
-            desc: '',
-            name: 'event',
-            type: 'Event'
-          }
-        ],
+        params: [{desc: '', name: 'event', type: 'Event'}],
         warnings: []
       }
     ]);
   });
 });
 
 suite('objectKeyToString', function() {
-  test('produces expected type names', function() {
-    const start = 0;
-    const end = 0;
-    const loc = {start: {line: 0, column: 0}, end: {line: 0, column: 0}};
-    const memberExpression: babel.MemberExpression = {
-      type: 'MemberExpression',
-      object: {
-        type: 'Identifier',
-        name: 'foo',
-        loc,
-        start,
-        end,
-      },
-      property: {
-        type: 'Identifier',
-        name: 'bar',
-        loc,
-        start,
-        end,
-      },
-      computed: false,
-      loc,
-      start,
-      end,
-    };
-    const afe: babel.ArrowFunctionExpression = {
-      id: {type: 'Identifier', name: 'foo', loc, start, end},
-      type: 'ArrowFunctionExpression',
-      expression: true,
-      generator: false,
-      async: false,
-      params: [],
-      body: {
-        type: 'Identifier',
-        name: 'foo',
-        loc,
-        start,
-        end,
-      },
-      loc,
-      start,
-      end,
-    };
-    const inputToOutput:
-        [
-          babel.Identifier|babel.StringLiteral|babel.NumericLiteral|
-          babel.MemberExpression|babel.ArrowFunctionExpression,
-          string|undefined
-        ][] =
-            [
-              [{type: 'Identifier', name: 'foo', loc, start, end}, 'foo'],
-              [
-                {
-                  type: 'StringLiteral',
-                  value: 'foo',
-                  loc,
-                  start,
-                  end,
-                  /* raw: '"foo"' */
-                },
-                'foo'
-              ],
-              [
-                {
-                  type: 'NumericLiteral',
-                  value: 10,
-                  loc,
-                  start,
-                  end,
-                  /* raw: '10' */
-                },
-                '10'
-              ],
-              [memberExpression, 'foo.bar'],
-              // When it hits an unknown type it returns undefined
-              [afe, undefined]
-            ];
-    for (const testCase of inputToOutput) {
-      assert.equal(testCase[1], objectKeyToString(testCase[0]));
+  test('produces expected names', function() {
+    const objectLiteralCode = `
+    ({
+      'foo': 1,
+      bar: 2,
+      [10]: 3,
+      [10 + 20]: 4,
+      ['hi' + ' there']: 5,
+      [identifier]: 6,
+    });
+    `;
+    const statement = babylon.parse(objectLiteralCode).program.body[0];
+    if (!babel.isExpressionStatement(statement)) {
```

Comment:
why not use the assert lib here?

Reply:
TypeScript can't reason about type narrowing with Assert